vumi_twilio_api/twilio_api.py

```diff
@@ -74,6 +222,66 @@ def __init__(self, message, format_='xml'):
         self.format_ = format_
 
 
+class Response(object):
+    """Base Response object used for HTTP responses"""
+    name = 'Response'
+
+    def __init__(self, **kw):
+        self._data = kw
+
+    def format_xml(self):
+        response = ET.Element("TwilioResponse")
+        root = ET.SubElement(response, self.name)
+
+        def format_xml_rec(dct, root):
+            for key, value in dct.iteritems():
+                if isinstance(value, dict):
+                    sub = ET.SubElement(root, key)
+                    format_xml_rec(value, sub)
+                else:
+                    sub = ET.SubElement(root, key)
+                    sub.text = value
+            return root
+
+        format_xml_rec(self._data, root)
+        return ET.tostring(response)
+
+    def format_json(self):
+        return json.dumps(convert_dict_keys(self._data))
+
+
+class Error(Response):
+    """Error HTTP response object, returned for incorred API queries"""
+    name = 'Error'
+
+    def __init__(self, error_type, error_message):
+        self._data = {
+            'error_type': error_type,
+            'error_message': error_message,
+        }
```

Comment:
We should probably doing something like ``super(Error, self).__init__(error_type=error_type, error_message=error_message)`` rather than setting ``_data``.